packages/ilmomasiina-backend/src/routes/signup/updateSignup.ts

```diff
@@ -50,26 +47,38 @@ export default async (
       throw new Forbidden('Signups closed for this event.');
     }
 
+    /** Is this signup already confirmed (i.e. is this the first update for this signup) */
+    const notConfirmedYet = !signup.confirmedAt;
     const questions = event.questions!;
 
     // Check that required common fields are present (if first time confirming)
-    if (!signup.confirmedAt) {
-      if (event.nameQuestion) {
-        if (!data.firstName) {
-          throw new BadRequest('Missing first name');
-        }
-        if (!data.lastName) {
-          throw new BadRequest('Missing last name');
-        }
+    const nameFields = (() => {
+      if (notConfirmedYet && event.nameQuestion) {
+        const { firstName, lastName } = request.body;
+        if (!firstName) { throw new BadRequest('Missing first name'); }
+        if (!lastName) { throw new BadRequest('Missing last name'); }
+        return { firstName, lastName };
       }
-      if (event.emailQuestion && !data.email) {
-        throw new BadRequest('Missing email');
+      return {};
+    })();
+
+    const emailField = (() => {
+      if (notConfirmedYet && event.emailQuestion) {
+        const { email } = request.body;
+        if (!email) { throw new BadRequest('Missing email'); }
+        return { email };
       }
-    }
+      return {};
+    })();
```

Comment:
I don't personally agree with the inline function trick - in my opinion a `let` with the modification nearby is much more readable. (This could also use a well-placed Lodash `pick`, IMO)

```suggestion
    let nameFields = {};
    if (notConfirmedYet && event.nameQuestion) {
      const { firstName, lastName } = request.body;
      if (!firstName) throw new BadRequest('Missing first name');
      if (!lastName) throw new BadRequest('Missing last name');
      nameFields = { firstName, lastName };
    }

    let emailField = {};
    if (notConfirmedYet && event.emailQuestion) {
      const { email } = request.body;
      if (!email) throw new BadRequest('Missing email');
      emailField = { email };
    }
```

Reply:
The original intention was to prevent accidents by defining nameFields and emailField as constants. But I agree using `let` makes it much more cleaner.
Suggested changes accepted in 8d88854.